src/platform/packages/shared/kbn-unified-metrics-grid/src/components/toolbar/hooks/use_toolbar_actions.tsx

```diff
@@ -34,11 +35,35 @@ export const useToolbarActions = ({
     onValuesChange,
     onClearValues,
     onClearAllDimensions,
+    onClearSearchTerm,
+    isFullscreen,
+    onToggleFullscreen,
+    searchTerm,
   } = useMetricsGridState();
 
+  const [showSearchInput, setShowSearchInput] = useState(false);
+
+  useDebounce(
+    () => {
+      setDebouncedSearchTerm(searchTerm);
```

Comment:
If I understood correct, we have a useState to hold this info. Shouldn't we have a redux state for this instead?